### ISSUE TITLE
IPNS very slow at solving the local node's name

### ISSUE THREAD

**Issue by fazo96** [ipfs/go-ipfs#2105]
Can we get some kind of boost with IPNS as far as the local node goes?

I mean, the `abc` node shouldn't take long to solve `/ipns/abc` but it does sometimes. This is on `0.4.0-dev`

This breaks the ability of any app that uses IPNS to store user data to work offline or behind scary big firewalls

**Comment by Faleidel** [comment]
Same issue as #2078 
`name publish` should add to the cache.

**Comment by lgierth** [comment]
Yep duplicate of #2078

**Comment by fazo96** [comment]
Uhm I'm not sure it's the same: I saw my node instantly resolve the name then after one minute or so struggling and taking a long time to resolve it. So it was in the cache for a while but it didn't stay there?

I'll try to see if the issue is still there after #2078

**Comment by yogster** [comment]
I can confirm this is still the case in 0.4.14. If I run `ipfs name resolve` takes about 1 minute to resolve, then resolves instantly for a minute, then takes another minute to resolve. I suppose the entry stays in the cache for a minute.

I host my blog in IPFS and have an nginx reverse proxy in my node to allow HTTPS access via my domain, rewriting requests to `ipns/domain`, but the way IPNS is at the moment this in unworkable. I'll have to either serve the blog from the local filesystem, or use the hash of my root folder for the time being.

Is there any workaround or config setting I can change to speed IPNS resolution of my own node up?

--- Hey by the way, thanks so much for your great work! I am prepping up to start contributing to this :-)

**Comment by fazo96** [comment]
@yogster I noticed that if you run the daemon with `ipfs daemon --enable-pubsub-experiment` IPNS is noticeably faster, even though I thought IPNS was not using pubsub yet

**Comment by yogster** [comment]
Hey @fazo96 thanks for replying so quickly! Sorry, I forgot to mention the reason I upgraded to 0.4.14 was so I could run the daemon with `--enable-pubsub-experiment`, as I was hoping it would speed up IPNS resolution... No luck.

Even if I run my daemon with `--enable-pubsub-experiment`, I've just run `ipfs config show` and noticed the following section doesn't mention pubsub:

```
  "Experimental": {
    "FilestoreEnabled": false,
    "Libp2pStreamMounting": false,
    "ShardingEnabled": false
  },
```
Is this normal? Is there anything I can do to diagnose this? Other than that I'm currently connected to 846 peers.

In any case (and bear in mind I haven't checked the inner workings of IPNS yet), shouldn't it be just a matter of checking the IPNS hash belongs to the local node, and returning the relevant IPFS hash? This should be irrelevant of whether PubSub or DHT are used, and should be instant.

Thanks again!
